scripts/update-go-mod/update-go-mod.py

```diff
@@ -0,0 +1,65 @@
+import os
+import re
+import sys
+
+def update_go_mod_file(go_mod_path, new_hash):
+    """
+    Update the go.mod file with the new commit hash.
+    """
+    try:
+        with open(go_mod_path, 'r') as file:
+            content = file.read()
+        # Print the original content for debugging
+        print("Original go.mod content:")
+        print(content)
+        updated_content = re.sub(r'(github\.com/multiversx/mx-chain-go\s+)[^\s]+', f'github.com/multiversx/mx-chain-go {new_hash}', content)
+
+        # Print updated content for verification
+        print("Updated go.mod content:")
+        print(updated_content)
+
+        # Write the updated content back to the file
+        with open(go_mod_path, 'w') as file:
+            file.write(updated_content)
+
+        print(f"Successfully updated go.mod with the latest commit hash: {new_hash}")
+    except Exception as e:
+        print(f"Failed to update go.mod file: {str(e)}")
+        sys.exit(1)
+
+def run_go_mod_tidy():
+    """
+    Run `go mod tidy` to ensure the go.mod file is tidy.
+    """
+    try:
+        print("Running go mod tidy...")
+        subprocess.run(['go', 'mod', 'tidy'], check=True)
+        print("Successfully ran go mod tidy.")
+
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to run go mod tidy: {e.stderr}")
+        sys.exit(1)
+
+if __name__ == "__main__":
```

Comment:
define a 'main' function and call this function in this section 

if __name__ == "__main__":
    main()